### PR TITLE
adding Task to install CLI on to agent

### PR DESCRIPTION
**Pull request by mozts2005**

### CODE REVIEW

**Comment by mozts2005** [comment]
@DavidStaheli ETA?

**Comment by davidstaheli** [comment]
This looks good to me.  @madhurig, would you or a team member like to review?

**Review by madhurig — APPROVED** [review]

File: Extension/CloudFoundryCLIInstall/utils.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+function getExecutableExtension(): string {
+    if (isWindows) {
+        return ".exe";
+    }
+    return "";
+}
+
+function getArchiveExtension(): string {
```
Comment (madhurig) [inline Extension/CloudFoundryCLIInstall/utils.ts:80]: This task fails on the Hosted macOS VMs. It appears that at the download URL for macOS https://packages.cloudfoundry.org/stable?release=macosx64&version=6.43.0, a .pkg is downloaded.

File: Extension/CloudFoundryCLIInstall/task.json [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    ],
+    "author": "Microsoft",
+    "demands": [],
+    "satisfies": [
+        "cf"
+    ],
+    "minimumAgentVersion": "2.142.1",
+    "version": {
+        "Major": 0,
+        "Minor": 1,
```
Comment (madhurig) [inline Extension/CloudFoundryCLIInstall/task.json]: Please change this to 150

File: Extension/extension-manifest.json [hunk 3]
```diff
@@ -2,12 +2,12 @@
     "manifestVersion": 1,
     "id": "cloud-foundry-build-extension",
     "name": "Cloud Foundry",
-    "version": "1.147.0",
+    "version": "1.148.0",
     "publisher": "ms-vsts",
```
Comment (madhurig) [inline Extension/extension-manifest.json:6]: Please change this to 1.150.0

File: externals.json [hunk 4]
```diff
@@ -1,3 +1,4 @@
 {
-    "azure-pipelines-task-lib":"2.7.7"
+    "azure-pipelines-task-lib":"2.7.7",
```
Comment (madhurig) [inline externals.json]: Switch to 2.8.0 which is the latest version

File: package.json [hunk 5]
```diff
@@ -39,7 +39,8 @@
     "through2": "^0.6.3",
     "typescript": "^2.4.1",
     "validator": "^3.33.0",
-    "azure-pipelines-task-lib": "^2.7.7"
+    "azure-pipelines-task-lib": "^2.7.7",
```
Comment (madhurig) [inline package.json]: Update to ^2.8.0

**Review by madhurig — CHANGES_REQUESTED** [review]
Please look at the failure for macOS, left a few more comments.

File: Extension/CloudFoundryCLIInstall/utils.ts (hunk 1 above)
Comment (mozts2005) [reply Extension/CloudFoundryCLIInstall/utils.ts:80]: I see what the issue is I will have it fixed later today

Comment (mozts2005) [reply Extension/CloudFoundryCLIInstall/utils.ts:80]: fix

File: Extension/CloudFoundryCLIInstall/task.json (hunk 2 above)
Comment (mozts2005) [reply Extension/CloudFoundryCLIInstall/task.json]: done

**Comment by mozts2005** [comment]
Updated to fix the macOS issue

**Review by madhurig — APPROVED** [review]
Looks good, thanks for the contribution.

**Pull request merged**